web_app/models/AC_model_iso9223.py

```diff
@@ -21,136 +21,105 @@ class ISO9223Model(CorrosionModel):
-
-        st.write(f'Corrosion speed selected: {self.parameters["corrosion_speed"]:.2f} μm/year')
-
-    def _handle_custom_corrosion_type(self) -> None:
-        """Handles custom corrosion types by prompting the user for detailed environmental inputs."""
-        with st.expander("Typical Values:"):
-            st.table(self.table_3)
-            st.write(
-                '### Parameters used in deriving the dose-response functions, including symbol, description, interval, and unit')
-            st.table(self.table_b3)
-            st.write(r'### Classification of contamination by sulfur-containing substances, represented by $SO_2$')
-            st.table(self.table_b4)
-            st.write(r'### Classification of contamination by sulfur-containing substances, represented by $Cl^-$')
-
-        limits = {
-            'T': {'desc': 'Temperature', 'lower': -17.1, 'upper': 28.7, 'unit': '°C'},
-            'RH': {'desc': 'Relative Humidity', 'lower': 34, 'upper': 93, 'unit': '%'},
-            'Pd': {'desc': 'SO₂-Deposit', 'lower': 0.7, 'upper': 150.4, 'unit': 'mg/(m²⋅d)'},
-            'Sd': {'desc': 'Cl⁻-Deposit', 'lower': 0.4, 'upper': 760.5, 'unit': 'mg/(m²⋅d)'}
-        }
-        for symbol, limit in limits.items():
-            value = st.text_input(f"Enter {limit['desc']} ({symbol}) [{limit['unit']}]:", value=limit['lower'])
-            self.parameters[symbol] = float(value)
-
-    def _determine_exponent(self) -> float:
-        """Determines or interpolates the exponent to be used in the material loss calculation."""
+
+            if selected_option == 'Use lower limit':
+                self.parameters['corrosion_speed'] = lower_limit
+            elif selected_option == 'Use upper limit':
+                self.parameters['corrosion_speed'] = upper_limit
+            else:
+                self.parameters['corrosion_speed'] = (lower_limit + upper_limit) / 2
+
+            st.write(f"Corrosion speed selected: {self.parameters['corrosion_speed']:.2f} μm/year")
+        else:
+            with st.expander("Typical Values:"):
+                st.table(self.table_3)
+                st.write("### Parameters used in deriving the dose-response functions")
+                st.table(self.table_b3)
+                st.write("### Classification of contamination by sulfur-containing substances, represented by SO₂ and Cl⁻")
+                st.table(self.table_b4)
+
+            limits = {
+                'T': {'desc': 'Temperature', 'lower': -17.1, 'upper': 28.7, 'unit': '°C'},
+                'RH': {'desc': 'Relative Humidity', 'lower': 34, 'upper': 93, 'unit': '%'},
+                'Pd': {'desc': 'SO₂-Deposit', 'lower': 0.7, 'upper': 150.4, 'unit': 'mg/(m²⋅d)'},
+                'Sd': {'desc': 'Cl⁻-Deposit', 'lower': 0.4, 'upper': 760.5, 'unit': 'mg/(m²⋅d)'}
+            }
+
+            for symbol, limit in limits.items():
+                value = st.number_input(f"Enter {limit['desc']} ({symbol}) [{limit['unit']}]:",
```

Comment:
I had a logic here to display an error message if the entered value is out of limits. Please check the old commits for it.

Reply:
This was removed in #12:
```
def get_input(symbol, limits):
    limit = limits[symbol]
    value = st.text_input(f"Enter {limit['desc']} ({symbol}) [{limit['unit']}]:", value=limit['lower'])

    if value:
        try:
            value = float(value)
            if value < limit['lower'] or value > limit['upper']:
                st.error(f"Please enter a value between {limit['lower']} and {limit['upper']} {limit['unit']}.")
            else:
                st.success(f"Value accepted: {value} {limit['unit']}")
        except ValueError:
            st.error("Please enter a valid number.")

    return value
```

Reply:
But we need this. We can't have the users entering invalid values.